extension/socks5proxy/bin/main.cpp

```diff
@@ -171,16 +186,47 @@ int main(int argc, char** argv) {
     return 1;
   }
 
-  Socks5* socks5 = new Socks5(config.port, config.addr, &app);
+  Socks5* socks5;
+  if (!config.localSocketName.isEmpty()) {
+    QLocalServer* server = new QLocalServer();
+    socks5 = new Socks5(server);
+    if (server->listen(config.localSocketName)) {
+      qDebug() << "Starting on local socket" << server->fullServerName();
+    } else if ((server->serverError() == QAbstractSocket::AddressInUseError) &&
+               QLocalServer::removeServer(config.localSocketName) &&
+               server->listen(config.localSocketName)) {
+      qDebug() << "(Re)starting on local socket" << server->fullServerName();
+    } else {
+      qWarning() << "Unable to listen to the local socket"
+                 << config.localSocketName;
+      qWarning() << "Listen failed:" << server->errorString();
+      return 1;
+    }
+  } else {
+    QTcpServer* server = new QTcpServer(socks5);
+    socks5 = new Socks5(server);
+    if (server->listen(config.addr, config.port)) {
+      qDebug() << "Starting on port" << config.port;
+    } else {
+      qWarning() << "Unable to listen to the proxy port" << config.port;
+      return 1;
+    }
+  }
+
   if (config.verbose) {
     startVerboseCLI(socks5);
   }
-  qDebug() << "Starting on port" << QString::number(config.port);
 
   QObject::connect(socks5, &Socks5::incomingConnection,
-                   [](const QString& peerAddress) {
-                     qDebug() << "Connection from  on port" << peerAddress;
+                   [](Socks5Connection* conn) {
+                     qDebug() << "Connection from peer" << conn->clientName();
                    });
 
+#ifdef __linux__
+  LinuxBypass* bypass = new LinuxBypass(socks5);
```

Comment:
because you don't use `bypass`, probably it's enough to do: `new LInuxByPass(socks5)` without setting the object.